cli/run/commandPlugins.ts

```diff
@@ -63,16 +60,34 @@ function loadAndRegisterPlugin(inputOptions: InputOptions, pluginText: string) {
 				if (pluginText[0] == '.') pluginText = path.resolve(pluginText);
 				plugin = require(pluginText);
 			} catch (ex) {
-				throw new Error(`Cannot load plugin "${pluginText}"`);
+				throw new Error(`Cannot load plugin "${pluginText}": ${ex.message}.`);
 			}
 		}
 	}
 	// some plugins do not use `module.exports` for their entry point,
 	// in which case we try the named default export and the plugin name
 	if (typeof plugin === 'object') {
-		plugin = plugin.default || plugin[pluginText]
+		plugin = plugin.default || plugin[getCamelizedPluginBaseName(pluginText)];
+	}
+	if (!plugin) {
+		throw new Error(
+			`Cannot find entry for plugin "${pluginText}". The plugin needs to export a function either as "default" or "${getCamelizedPluginBaseName(
+				pluginText
+			)}" for Rollup to recognize it.`
+		);
 	}
 	inputOptions.plugins!.push(
 		typeof plugin === 'function' ? plugin.call(plugin, pluginArg) : plugin
 	);
 }
+
+function getCamelizedPluginBaseName(pluginText: string): string {
+	return pluginText
+		.match(/^(@rollup\/plugin-|rollup-plugin-)?(.*)$/)![2]
```

Comment:
If the `^` is removed from the regex then paths such as the following could be correctly handled:
```
./node_modules/@rollup/plugin-node-resolve
/absolute/dir/rollup-plugin-node-resolve
```
Also, replacing `(.*)` with `([-\w]+)` would be better - has at least one non-dash, non-space character.

Reply:
I'd rather leave filesystem resolution up to node. if we assert a package name, users could alias that package to their file system using package.json

Reply:
Ignore the `node_modules` and `node-resolve` in the examples - they could be anything. My comment is unrelated to node file resolution. It solely a `--plugin` parameter parsing issue. The CLI plugin parameter already supports relative and absolute path arguments. Look at the tests for examples.

Reply:
Makes sense. I will add some changes.

Reply:
lgtm